exiftool/exiftool.py

```diff
@@ -530,25 +539,29 @@ def execute_json(self, *params):
 	def get_metadata_batch_wrapper(self, filenames, params=None):
 		return self.execute_json_wrapper(filenames=filenames, params=params)
 
-	def get_metadata_batch(self, filenames):
+	def get_metadata_batch(self, filenames, params=None):
 		"""Return all meta-data for the given files.
 
 		The return value will have the format described in the
 		documentation of :py:meth:`execute_json()`.
 		"""
-		return self.execute_json(*filenames)
+		if not params:
+			params = []
+		return self.execute_json(*filenames, *params)
 
 	# (#11)
 	def get_metadata_wrapper(self, filename, params=None):
 		return self.execute_json_wrapper(filenames=[filename], params=params)[0]
 
-	def get_metadata(self, filename):
+	def get_metadata(self, filename, params=None):
 		"""Return meta-data for a single file.
 
 		The returned dictionary has the format described in the
 		documentation of :py:meth:`execute_json()`.
 		"""
-		return self.execute_json(filename)[0]
+		if not params:
+			params = None
```

Comment:
is this redundant?  If **not params**, params is already None

Reply:
It has been awhile so you may be right. I think there was an issue with non-none false (empty lists) values being passed to execute_json.

Reply:
I checked out my local copy, this should be params = [] to prevent trying to unpack "none"